src/languages/gml.js

```diff
@@ -2994,135 +898,569 @@ export default function(hljs) {
+  /**
+   * Regex for a dot separating some LHS and RHS expression with optional whitespace (as this is
+   * supported in the engine.)
+   */
+  const DOT_ACCESSOR_RE = /\b\.\b/;
+
+  /**
+   * Expressions, which form part of a valid statement.
+   * @type {Object[]}
+   */
+  const EXPRESSION = [];
+
+  /**
+   * A single-line comment.
+   */
+  const COMMENT_LINE = hljs.COMMENT('//', /$/);
+
+  const JSDOC_TYPE_PARAM_RE = /{ *[a-zA-Z_][a-zA-Z0-9_\.<>\|]* *} */;
+
+  const JSDOC_ANNOTATIONS = [
+    {
+      match: /@((desc(ription)?)|ignore|pure|deprecated|(func(tion)?))\b/,
+      scope: "doctag"
+    },
+    {
+      match: [
+        /@(self|context) */,
+        JSDOC_TYPE_PARAM_RE
+      ],
+      scope: {
+        1: "doctag",
+        2: "type",
+      }
+    },
+    {
+      match: [
+        /(@return)s? */,
+        JSDOC_TYPE_PARAM_RE
+      ],
+      scope: {
+        1: "doctag",
+        2: "type",
+      },
+      relevance: 0,
+    },
+    {
+      match: [
+        /@((param(eter)?)|arg(ument)?) */,
+        JSDOC_TYPE_PARAM_RE,
+        /\[?/,
+        VALID_IDENTIFIER_RE,
+        /(( *=[^\n]+?)?\])? *?/
+      ],
+      scope: {
+        1: "doctag",
+        2: "type",
+        3: "params",
+        4: "params",
+        5: "params",
+      },
+      relevance: 0,
+    }
+  ];
+
+  /**
+   * A comment that documents a function using the same style as JSDoc for JavaScript.
+   */
+  const COMMENT_JSDOC = {
+    begin: /\/\/\//,
+    end: /$/,
+    scope: "comment",
+    relevance: 0,
+    contains: JSDOC_ANNOTATIONS,
+  };
+
+  /**
+   * A comment that documents a function using the same style as JSDoc for JavaScript.
+   */
+  const COMMENT_BLOCK_JSDOC = {
+    begin: /\/\*\*/,
+    end: /\*\//,
+    scope: "comment",
+    relevance: 0,
+    contains: JSDOC_ANNOTATIONS
+  };
+
+  /**
+   * Modes for the types of comments supported in GML.
+   */
+  const COMMENT = {
+    variants: [
+      COMMENT_JSDOC,
+      COMMENT_BLOCK_JSDOC,
+      COMMENT_LINE,
+      hljs.C_BLOCK_COMMENT_MODE,
+    ]
+  };
+
+  /**
+   * A template string substitution. `contains` is filled in after `EXPRESSION` is defined due to
+   * nesting.
+   */
+  const STRING_SUBSTITUTION = {
+    begin: /{/,
+    end: /}/,
+    beginScope: "char.escape",
+    endScope: "char.escape",
+    scope: "subst",
+    contains: EXPRESSION
+  };
+
+  /**
+   * A template string substitution for use with the older `string()` optional args with `"{0}"`,
+   * etc.
+   */
+  const STRING_NUMERICAL_SUBSTITUTION = {
+    match: /{[0-9]+}/,
+    scope: "subst"
+  };
+
+  /**
+   * An escape sequence in a string.
+   */
+  const STRING_ESCAPE = {
+    scope: "char.escape",
+    variants: [
+      { match: /\\u[a-fA-F0-9]{1,6}/ },
+      { match: /\\[^\n]/ }
+    ]
+  };
+
+  /**
+   * Various types of strings supported in the engine.
+   */
+  const STRING = {
+    variants: [
+      {
+        begin: /\$"/,
+        end: /\"/,
```

Comment:
Is this \ an escape?  Do we need to escape `"`?

Reply:
The escape is to escape the functionality of `$`, since this is matching the literal text `$"`, as GML's template strings/string interpolation is in the form `$"some text with {some_variable}, etc"`. They are effectively identical to C#'s string interpolation.